src/Shipping/PickupLocation.php

```diff
@@ -0,0 +1,361 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Shipping;
+
+use WC_Shipping_Method;
+
+/**
+ * Local Pickup Shipping Method.
+ */
+class PickupLocation extends WC_Shipping_Method {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->id                 = 'pickup_location';
+		$this->method_title       = __( 'Pickup Locations', 'woo-gutenberg-products-block' );
+		$this->method_description = __( 'Allow customers to choose a local pickup location during checkout.', 'woo-gutenberg-products-block' );
+		$this->init();
+	}
+
+	/**
+	 * Init function.
+	 */
+	public function init() {
+		$this->init_form_fields();
+		$this->init_settings();
+
+		$this->enabled          = $this->get_option( 'enabled' );
+		$this->title            = $this->get_option( 'title' );
+		$this->tax_status       = $this->get_option( 'tax_status' );
+		$this->cost             = $this->get_option( 'cost' );
```

Comment:
Nothing would break. I just left it out because of the designs. We just need to add it to the UI. Another follow up - I'd suggest getting design feedback too since we didn't implement the modal.